code/modules/mob/living/carbon/human/species/species.dm

```diff
@@ -249,77 +301,68 @@
+			H.failed_last_breath = 0
+			H.adjustOxyLoss(-5)
+			CO2_used = breath.carbon_dioxide / 6
 
-	if(Toxins_pp > safe_toxins_max) // Too much toxins
-		var/ratio = (breath.toxins/safe_toxins_max) * 10
-		//adjustToxLoss(Clamp(ratio, MIN_PLASMA_DAMAGE, MAX_PLASMA_DAMAGE))	//Limit amount of damage toxin exposure can do per second
-		if(H.reagents)
-			H.reagents.add_reagent("plasma", Clamp(ratio, MIN_PLASMA_DAMAGE, MAX_PLASMA_DAMAGE))
-		H.toxins_alert = max(H.toxins_alert, 1)
+	breath.oxygen   		-= O2_used
+	breath.nitrogen 		-= N2_used
+	breath.toxins   		-= Tox_used
+	breath.carbon_dioxide 	-= CO2_used
+	breath.carbon_dioxide 	+= O2_used
 
-	else if(O2_pp > vox_oxygen_max && name == "Vox") //Oxygen is toxic to vox.
-		var/ratio = (breath.oxygen/vox_oxygen_max) * 1000
-		H.adjustToxLoss(Clamp(ratio, MIN_PLASMA_DAMAGE, MAX_PLASMA_DAMAGE))
-		H.toxins_alert = max(H.toxins_alert, 1)
-	else
-		H.toxins_alert = 0
 
 	if(breath.trace_gases.len)	// If there's some other shit in the air lets deal with it here.
 		for(var/datum/gas/sleeping_agent/SA in breath.trace_gases)
-			var/SA_pp = (SA.moles/breath.total_moles())*breath_pressure
-			if(SA_pp > SA_para_min) // Enough to make us paralysed for a bit
+			var/SA_pp = (SA.moles / breath.total_moles()) * breath_pressure
+			if(SA_pp > atmos_requirements["sa_para"]) // Enough to make us paralysed for a bit
 				H.Paralyse(3) // 3 gives them one second to wake up and run away a bit!
-				if(SA_pp > SA_sleep_min) // Enough to make us sleep as well
-					H.sleeping = max(H.sleeping+2, 10)
+				if(SA_pp > atmos_requirements["sa_sleep"]) // Enough to make us sleep as well
+					H.sleeping = max(H.sleeping + 2, 10)
 			else if(SA_pp > 0.01)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
 				if(prob(20))
-					spawn(0) H.emote(pick("giggle", "laugh"))
+					spawn(0)
+						H.emote(pick("giggle", "laugh"))
 
 	handle_temperature(breath, H)
-
 	return 1
 
 /datum/species/proc/handle_temperature(datum/gas_mixture/breath, var/mob/living/carbon/human/H) // called by human/life, handles temperatures
 	if( (abs(310.15 - breath.temperature) > 50) && !(RESIST_HEAT in H.mutations)) // Hot air hurts :(
 		if(H.status_flags & GODMODE)	return 1	//godmode
 		if(breath.temperature < cold_level_1)
 			if(prob(20))
-				to_chat(H, "\red You feel your face freezing and an icicle forming in your lungs!")
+				to_chat(H, "<span class='warning'>You feel your face freezing and an icicle forming in your lungs!</span>")
 		else if(breath.temperature > heat_level_1)
 			if(prob(20))
-				to_chat(H, "\red You feel your face burning and a searing heat in your lungs!")
+				to_chat(H, "<span class='warning'>You feel your face burning and a searing heat in your lungs!</span>")
 
 
 
 		switch(breath.temperature)
 			if(-INFINITY to cold_level_3)
 				H.apply_damage(cold_env_multiplier*COLD_GAS_DAMAGE_LEVEL_3, BURN, "head", used_weapon = "Excessive Cold")
-				H.fire_alert = max(H.fire_alert, 1)
```

Comment:
these aren't a thing on -tg-, it's all handled in body temperature code on the human